tests/shared/src/test/scala/spire/math/RealScalaCheckSuite.scala

```diff
@@ -112,25 +114,28 @@ class RealScalaCheckSuite extends munit.ScalaCheckSuite {
 
   property("x.pow(3) = x * x * x") {
     forAll { (x: Real) =>
-      x.pow(2) == x * x
+      x.pow(3) == x * x * x
     }
   }
 
   property("x.pow(k).nroot(k) = x") {
+    implicit val realArb: Arbitrary[Real] = Arbitrary(realFromLongs)
     forAll { (x0: Real, k: Sized[Int, _1, _10]) =>
```

Comment:
I'm not really sure if this is any nicer, but it is a bit more explicit :)

```suggestion
    forAll(realFromLongs, arbitrary[Sized[Int, _1, _10]]) { (x0, k) =>
```

Reply:
Ah, this will fix the compile errors I reckon.

Reply:
Yeah, this worked in 3 but not in 2 😮 -- implicit resolution must have changed in 3.
I split these tests into a separate suite so that it's more clear.

Reply:
I think if we explicitly pass the generator, then we can keep them in the same file.

Reply:
Happy to go that route if that's standard here - I felt kind of bad writing `arbitrary[Sized[Int, _1, _10]]` 3 times though 🤷‍♂️

Reply:
I'm not entirely sure what `Sized` is, but you may be able to use `Gen.chooseNum` or something to achieve the same result. I think it's worth it to keep the tests together.

Reply:
I'm struggling to find a solution that uses `realFromLongs` in the derivation of `Arbitrary[NonZero[Real]]` for [this case](https://github.com/typelevel/spire/pull/1173/files#diff-f8231a49bbd9e332a3e1bd93a9fdce1b9903cd499b2bf7c6c4fd0b7cda38d107R46) that is compatible between scala 2 and 3 but keeps everything in one suite. It seems like where I'd previously have used an `implicit def` in scala 2, [I should be using a `given` in scala 3](https://docs.scala-lang.org/scala3/reference/contextual/relationship-implicits.html#given-instances). I suppose two suites in one file would be doable here too, but that's not far off from what's already in the PR.

I pulled in that test here because it seemed like it was susceptible to the same problem in the issue. What do you think the path forward should be?

Reply:
I'm not sure I understand what the Scala 2/3 issue is here, since `implicit def` should also work on Scala 3. Are you saying that `Arbitrary[NonZero[Real]]` is not using the `realFromLongs` at the top of that file?

Reply:
Ok so I read the compiler output again and it's not a compat issue, I'm just not sure how to provide implicit evidence properly. Sorry about the confusion.

Here's the current `NonZero` instance:
```scala
 implicit def nonZeroSpireImplicit[A: Signed: AdditiveGroup: Arbitrary]: Arbitrary[NonZero[A]] =
    Arbitrary(arbitrary[A].filter(_.signum != 0).map(NonZero(_)))
```

I want the instance of `Arbitrary[A]` used here to be `realFromLongs`, so I updated it to:
```scala
implicit def nonZeroSpireImplicit[A: Signed: AdditiveGroup](implicit arbA: Arbitrary[A]): Arbitrary[NonZero[A]] =
    Arbitrary(arbA.arbitrary.filter(_.signum != 0).map(NonZero(_)))
```
I was hoping to pass in `realFromLongs` explicitly, but in doing so I can't find evidence for `Signed[Real]`.

Reply:
Aha, gotcha, no worries :) IIRC before you changed it the signature was like this:
```scala
implicit def nonZeroSpireImplicit[A: Signed: AdditiveGroup: Arbitrary]: Arbitrary[NonZero[A]]
```

You can then explicitly pass in `realFromLongs` something like this:
```scala
nonZeroSpireImplicit[Real](implicitly, implicitly, Arbitrary(realFromLongs))
```